### PR TITLE
✨ Change requestResize interface to return promise

### PR DESCRIPTION
**Pull request by powerivq**
Changing the requestResize API to return promise. This depends on https://github.com/ampproject/amphtml/pull/29532 to function properly on older browsers.

### CODE REVIEW

File: 3p/ampcontext.js [hunk 1]
```diff
@@ -107,13 +107,20 @@ export class AbstractAmpContext {
     /** @type {?string} */
     this.tagName = null;
 
+    /** @type {!Object<number, Array<function>>} */
+    this.resizeIdToHandlers_ = new Object();
```
Comment (calebcordry) [inline 3p/ampcontext.js:111]: nit: map

File: 3p/ampcontext.js [hunk 2]
```diff
@@ -210,16 +217,43 @@ export class AbstractAmpContext {
...
     this.client_.sendMessage(
       MessageType.EMBED_SIZE,
       dict({
+        'id': requestId,
         'width': width,
         'height': height,
         'hasOverflow': hasOverflow,
       })
     );
+    return new Promise((resolve, reject) => {
```
Comment (calebcordry) [inline 3p/ampcontext.js:233]: nit: Deferred

File: 3p/ampcontext.js [hunk 3]
```diff
@@ -210,16 +217,43 @@ export class AbstractAmpContext {
...
+  }
+
+  /**
+   *  Set up listeners to handle responses from request size.
+   */
+  listenToResizeResponse_() {
+    this.client_.registerCallback(MessageType.EMBED_SIZE_CHANGED, (data) => {
+      const id = data['id'];
+      if (id !== undefined) {
+        this.resizeIdToHandlers_[id][0]();
```
Comment (calebcordry) [inline 3p/ampcontext.js:245]: nit: maybe name these so it's easier to understand what is happening. eg. `this.resizeIdToHandlers_[id].resolve()`

File: extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js [hunk 4]
```diff
@@ -170,6 +170,7 @@ export class AmpAdXOriginIframeHandler {
             this.element_.warnOnMissingOverflow = false;
           }
           this.handleResize_(
+            data['id'],
```
Comment (calebcordry) [inline extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js:173]: I'm not very familiar with this code. Can you explain when this has an id or not, and why there is no id for `renderStartMsgHandler_`?

Comment (powerivq) [reply extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js:173]: `renderStartMsgHandler_` would send a resize_success event upon RENDER_START, therefore it is not associated with any resize requests and no resize request ID is available.

Comment (calebcordry) [reply extensions/amp-ad/0.1/amp-ad-xorigin-iframe-handler.js:173]: Got it. Thanks!

**Pull request merged**